packages/core/src/cli/commands/build.command.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  copyFile, rm, cp, writeFile,
+} from 'fs/promises';
+import { exec } from 'child_process';
+import { rollup, RollupBuild } from 'rollup';
+import { nodeResolve } from '@rollup/plugin-node-resolve';
+import commonjsRollupPlugin from '@rollup/plugin-commonjs';
+// import typescriptRollupPlugin from '@rollup/plugin-typescript';
+import sucraseRollupPlugin from '@rollup/plugin-sucrase';
+
+import kleur from 'kleur';
+import _ from 'lodash-es';
+import { DmnoCommand } from '../lib/dmno-command';
+
+import { addServiceSelection } from '../lib/selection-helpers';
+import { getCliRunCtx } from '../lib/cli-ctx';
+import { addCacheFlags } from '../lib/cache-helpers';
+import { addWatchMode } from '../lib/watch-mode-helpers';
+import { checkForSchemaErrors } from '../lib/check-errors-helpers';
+import { CliExitError } from '../lib/cli-error';
+import { DmnoBuildInfo } from '../../config-loader/find-services';
+
+const program = new DmnoCommand('build')
+  .summary('build compiled dmno config')
+  .description('Builds DMNO config files into minimal code appropriate for deployed contexts without access to all dependencies')
+  .example('dmno build', 'Build the config using current detected service')
+  .example('dmno build --service service1', 'Build dmno config files for use in service1');
+
+addWatchMode(program); // must be first
+addCacheFlags(program);
+addServiceSelection(program);
+
+program.action(async (opts: {
+  // these args should be handled already by the helpers
+  // service?: string,
+  // watch?: boolean,
+  // skipCache?: boolean,
+  // clearCache?: boolean,
+
+  format?: string,
+  public?: boolean,
+  showAll?: boolean,
+}, thisCommand) => {
+  const ctx = getCliRunCtx();
+
+  if (opts.format) ctx.expectingOutput = true;
+
+  if (!ctx.selectedService) return; // error message already handled
+
+  ctx.log(`\nResolving config for service ${kleur.magenta(ctx.selectedService.serviceName)}\n`);
+
+  const workspace = ctx.workspace!;
+  const service = ctx.selectedService;
+  checkForSchemaErrors(workspace);
+
+  if (!ctx.configLoader.workspaceInfo.isMonorepo) {
+    throw new CliExitError('`dmno build` only works in monorepos');
```

Comment:
maybe something slightly more explicit here, like "`dmno build` is only necessary in monorepos`, please use `dmno run`"

Reply:
maybe just link to docs page